tools/gke-gpu-driver-version/src/main.rs

```diff
@@ -0,0 +1,114 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Ok;
+use chrono::Utc;
+use clap::Parser;
+use env_logger::Env;
+use log::error;
+
+use crate::pb::GpuDriverVersionInfoList;
+
+tonic::include_proto!("cos_tools");
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    env_logger::Builder::from_env(Env::default().default_filter_or("info"))
+        .format_timestamp(None)
+        .init();
+
+    let gke_to_cos_versions =
+        reqwest::get("https://www.gstatic.com/gke-image-maps/gke-to-cos.json")
+            .await?
+            .json::<GkeToCos>()
+            .await?;
+
+    let args = Args::parse();
+
+    let cos_version = gke_to_cos_versions
+        .entries
+        .into_iter()
+        .find(|v| args.cluster_version.eq(&v.gke_version))
+        .map(|v| v.image);
+
+    match cos_version {
+        Some(v) => {
+            let mut data = v.rsplit("-").into_iter().take(3).collect::<Vec<_>>();
+
+            data.reverse();
+
+            let version = data.join(".");
+            let proto = reqwest::get(format!(
+                "https://storage.googleapis.com/cos-tools/{}/lakitu/gpu_driver_versions.textproto",
+                version
+            ))
+            .await?
+            .text()
+            .await?;
+
+            let list: GpuDriverVersionInfoList = serde_yml::from_str(&proto)?;
+            let mut result = GpuVersionData::default();
+            result.creation_time = gke_to_cos_versions.creation_time;
+            result.cos_version = v;
+            list.gpu_driver_version_info
+                .into_iter()
+                .filter(|v| match &v.gpu_device {
+                    Some(device) => device.gpu_type == args.gpu_type,
+                    None => false,
+                })
+                .flat_map(|v| v.supported_driver_versions)
+                .for_each(|v| {
+                    if v.label == "DEFAULT" {
+                        result.default_driver_version = v.version.clone();
+                    }
+                    if v.label == "LATEST" {
+                        result.latest_driver_version = v.version.clone();
+                    }
+                });
+            println!("{}", serde_json::to_string_pretty(&result)?);
+        }
+        None => error!("cos version not found for cluster version"),
+    }
+
+    Ok(())
+}
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    #[arg(long, default_value = "")]
+    cluster_version: String,
```

Comment:
```suggestion
    gke_version: String,
```

Technically it's more a GKE version right? When I think of a "cluster version" I think of a control plane version. However, tool really just applies to GKE node versions, right?

So I guess `gke_node_version` could also work...

Reply:
I see what you're saying, changed it to `gke_version`, to better match the tool name.